capi/src/visitors/fragment-tags.ts

```diff
@@ -1,13 +1,25 @@
 import * as t from "../types";
 import clone from "clone-deep";
 
+const isPContainingFragments = function(node: t.HyperscriptNode): boolean {
+  if (!Array.isArray(node)) return false;
+  if (node[0] !== "p") return false;
```

Comment:
So this is writing to the logic that the first node will always be a `div` ? Is there a chance it can be something different like a header tag or anything?

Reply:
This code checks to see if the hyperscript node we're looking at is a `<p>` containing `<internal-fragment>`s.  If it is, Stencil freaks out for some unknown reason (that I can't reproduce locally with a simpler example...) but that seems to be related to this expected behavior where elements aren't supposed to be nested in `<p>`s: https://codesandbox.io/s/div-in-a-p-forked-d5q1e.